src/solution.c

```diff
@@ -468,12 +472,49 @@ static void solution_thread(void *arg)
       continue;
     }
 
+    /* check if we have a solution, if yes calc iono correction */
+    if (soln_flag) {
+      for (u8 i = 0; i < n_ready_tdcp; i++) {
+        ionosphere_t iono_params;
+        double az, el, iono_correction, tropo_correction, cp_correction;
+        /* read iono parameters for the tracking channel if available */
+        if (gps_iono_params_read(&iono_params)) {
```

Comment:
This check should only wrap `calc_ionosphere` as `calc_troposphere` does not need the iono params.